src/node/history.h

```diff
@@ -678,7 +680,8 @@ namespace ccf
     {
       std::lock_guard<ccf::pal::Mutex> guard(state_lock);
       return replicated_state_tree.serialise(
-        replicated_state_tree.begin_index(), to);
+        replicated_state_tree.begin_index(),
+        std::min(to, replicated_state_tree.end_index()));
```

Comment:
This behaviour (silently serialising a different tree than was requested) feels dangerous. I assume someone up-the-stack is discarding the return value if `to` is no longer available - can we return an empty response in that case instead?